runtime/src/bank.rs

```diff
@@ -6648,6 +6649,111 @@ impl Bank {
+        (
+            offset,
+            end_offset,
+            solana_sha256_hasher::hash(buffer_program_data),
+        )
+    }
+
+    fn enshrine_slashing_program(&mut self) {
+        const SLASHING_PROGRAM_KEY: Pubkey =
+            Pubkey::from_str_const("S1ashing11111111111111111111111111111111111");
+        const BUFFER_PROGRAM_KEY: Pubkey =
+            Pubkey::from_str_const("S1asHs4je6wPb2kWiHqNNdpNRiDaBEDQyfyCThhsrgv");
+        // 192ed727334abe822d5accba8b886e25f88b03c76973c2e7290cfb55b9e1115f
+        const HASH_BYTES: [u8; 32] = [
+            0x19, 0x2e, 0xd7, 0x27, 0x33, 0x4a, 0xbe, 0x82, 0x2d, 0x5a, 0xcc, 0xba, 0x8b, 0x88,
+            0x6e, 0x25, 0xf8, 0x8b, 0x03, 0xc7, 0x69, 0x73, 0xc2, 0xe7, 0x29, 0x0c, 0xfb, 0x55,
+            0xb9, 0xe1, 0x11, 0x5f,
+        ];
+        const VERIFIED_BUILD_HASH: Hash = Hash::new_from_array(HASH_BYTES);
+
+        let slashing_program_data_key =
+            bpf_loader_upgradeable::get_program_data_address(&SLASHING_PROGRAM_KEY);
+        let buffer_program_data_key =
+            bpf_loader_upgradeable::get_program_data_address(&BUFFER_PROGRAM_KEY);
+
+        if self
+            .get_account(&SLASHING_PROGRAM_KEY)
+            .map(|acc| !acc.data().is_empty())
+            .unwrap_or(false)
+        {
+            error!("Slashing program account is already populated");
+            return;
+        }
+
+        if self
+            .get_account(&slashing_program_data_key)
+            .map(|acc| !acc.data().is_empty())
+            .unwrap_or(false)
+        {
+            error!("Slashing program data account is already populated");
+            return;
+        }
+
+        let Some(buffer_program_account) = self.get_account(&BUFFER_PROGRAM_KEY) else {
+            error!("Buffer program account {} is missing", BUFFER_PROGRAM_KEY);
+            return;
+        };
+
+        let Some(buffer_program_data_account) = self.get_account(&buffer_program_data_key) else {
+            error!(
+                "Buffer program data account {} is missing",
+                buffer_program_data_key
+            );
+            return;
+        };
+
+        let (offset, end_offset, build_hash) =
+            Self::get_build_hash_and_offsets(buffer_program_data_account.data());
+        if build_hash != VERIFIED_BUILD_HASH {
+            error!(
+                "Buffer verified build hash {} does not match expected {}",
+                build_hash, VERIFIED_BUILD_HASH
+            );
+            return;
+        }
+
+        let mut slashing_program_account = buffer_program_account.clone();
+        slashing_program_account.set_owner(bpf_loader_upgradeable::id());
+        slashing_program_account
+            .set_state(&bpf_loader_upgradeable::UpgradeableLoaderState::Program {
+                programdata_address: slashing_program_data_key,
+            })
+            .unwrap();
+        slashing_program_account.set_executable(true);
```

Comment:
nit: Are the `set_owner` and `set_executable` steps necessary since it's being cloned from `buffer_program_account`?